### PR TITLE
Make sure using default host doesn't raise error

### LINKED ISSUE #53: Using default host variable results in "Can't modify frozen array" error
I tried to use the `QueuePublisher` (not just related to the `QueuePublisher` though) in a test in wally but got the following message. The reason is that I used the default host variable, which is now frozen since https://github.com/twingly/twingly-amqp/commit/22acd31ccd2263351a1dd8df5067ebdc8b3fba3f#diff-a37f3e6add223b058bddda335dda3b17R10 (❤️  RuboCop). Everything works when using a `RABBITMQ_01_HOST` env variable, since the array isn't frozen then.

The line in bunny that causes the problem: https://github.com/ruby-amqp/bunny/blob/4c4a968c287995c013c0ca8a383fce34bb22e0b9/lib/bunny/session.rb#L774

```
     RuntimeError:
       can't modify frozen Array
     # /Users/mattias/.gem/ruby/2.2.4/gems/bunny-2.5.1/lib/bunny/session.rb:774:in `map!'
     # /Users/mattias/.gem/ruby/2.2.4/gems/bunny-2.5.1/lib/bunny/session.rb:774:in `addresses_from'
     # /Users/mattias/.gem/ruby/2.2.4/gems/bunny-2.5.1/lib/bunny/session.rb:148:in `initialize'
     # /Users/mattias/.gem/ruby/2.2.4/gems/bunny-2.5.1/lib/bunny.rb:66:in `new'
     # /Users/mattias/.gem/ruby/2.2.4/gems/bunny-2.5.1/lib/bunny.rb:66:in `new'
     # /Users/mattias/.gem/ruby/2.2.4/bundler/gems/twingly-amqp-fd728d68cd93/lib/twingly/amqp/session.rb:20:in `create_connection'
     # /Users/mattias/.gem/ruby/2.2.4/bundler/gems/twingly-amqp-fd728d68cd93/lib/twingly/amqp/session.rb:14:in `initialize'
     # /Users/mattias/.gem/ruby/2.2.4/bundler/gems/twingly-amqp-fd728d68cd93/lib/twingly/amqp/connection.rb:20:in `new'
     # /Users/mattias/.gem/ruby/2.2.4/bundler/gems/twingly-amqp-fd728d68cd93/lib/twingly/amqp/connection.rb:20:in `block in instance'
     # /Users/mattias/.gem/ruby/2.2.4/bundler/gems/twingly-amqp-fd728d68cd93/lib/twingly/amqp/connection.rb:18:in `synchronize'
     # /Users/mattias/.gem/ruby/2.2.4/bundler/gems/twingly-amqp-fd728d68cd93/lib/twingly/amqp/connection.rb:18:in `instance'
     # /Users/mattias/.gem/ruby/2.2.4/bundler/gems/twingly-amqp-fd728d68cd93/lib/twingly/amqp/queue_publisher.rb:11:in `initialize'
```

### PR DESCRIPTION
**Pull request by roback**
fix #53

### CODE REVIEW

File: lib/twingly/amqp/session.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@ def hosts_from_env
         end
         hosts = environment_keys_with_host.map { |key| ENV[key] }
 
-        return DEFAULT_HOSTS if hosts.empty?
+        return DEFAULT_HOSTS.dup if hosts.empty?
```
Comment (walro) [inline lib/twingly/amqp/session.rb]: How about we always dup hosts before passing to bunny?

Comment (walro) [inline lib/twingly/amqp/session.rb]: Not really related to this PR, but I think I'd prefer 

```ruby
hosts = DEFAULT_HOSTS if hosts.empty?
```

to just have one return path here

Comment (roback) [inline lib/twingly/amqp/session.rb]: > How about we always dup hosts before passing to bunny?

Yes. That would also fix the problem where we pass a frozen host array to `Session.new` ourselves:

```ruby
it "should not raise an error when given frozen array of hosts" do
  expect do
    described_class.new(hosts: ["localhost"].freeze)
  end.not_to raise_error
end

#    Failure/Error: expect do
#       expected no Exception, got #<RuntimeError: can't modify frozen Array> with backtrace:
# ...
```

File: lib/twingly/amqp/session.rb [hunk 2]
```diff
@@ -27,6 +27,8 @@ def connection_options
         options[:pass]  ||= password_from_env
         options[:hosts] ||= hosts_from_env
 
+        options[:hosts] = options.fetch(:hosts).dup
```
Comment (roback) [inline lib/twingly/amqp/session.rb]: Any improvement suggestions on this is welcome since I think it looks a bit weird :)

Comment (walro) [inline lib/twingly/amqp/session.rb]: how about duping the whole options array instead? 

`default_connection_options.merge(options.dup)`

Comment (roback) [inline lib/twingly/amqp/session.rb]: I had the same idea myself, but unfortunately it didn't work:

```ruby
{hosts: ["localhost"].freeze}.dup.tap { |h| h[:hosts].map! { "test" } }
# RuntimeError: can't modify frozen Array
# from (pry):1:in `map!'
```

Comment (roback) [inline lib/twingly/amqp/session.rb]: Sure, I'll add a comment. The issue at bunny is closed now, but its not released yet.

Comment (walro) [inline lib/twingly/amqp/session.rb]: maybe make a follow-up issue here so we can remove it once bunny has released (we will need to change our gemspec too, to require at least that version of bunny).

File: spec/integration/twingly/amqp/session_spec.rb [hunk 3]
```diff
@@ -22,6 +22,31 @@
...
+          spec.run
+
+          ENV["RABBITMQ_01_HOST"] = host
+        end
+
+        let(:default_host) { { hosts: ["localhost"] } }
+
+        it "should use the default host" do
+          expect(Bunny)
+          .to receive(:new)
```
Comment (walro) [inline spec/integration/twingly/amqp/session_spec.rb]: indent is off here, two spaces lacking for the chained calls

File: lib/twingly/amqp/session.rb (hunk 2 above)
Comment (roback) [inline lib/twingly/amqp/session.rb]: Opened #57 so we wont forget about it :)

File: spec/integration/twingly/amqp/session_spec.rb (hunk 3 above)
Comment (roback) [inline spec/integration/twingly/amqp/session_spec.rb]: I ran RuboCop to see if there were any more mistakes, but that was a mistake :P

```ruby
lib/twingly/amqp/session.rb:25:7: C: Assignment Branch Condition size for connection_options is too high. [16.49/15]
      def connection_options
      ^^^
```

**Pull request merged**